File: 209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        ans = 10**9 + 1
        l, r = 0, 0
        curr = 0
        for r in range(len(nums)):
            curr += nums[r]
            while curr >= target:
                curr -= nums[l]
                ans = min(r - l + 1, ans)
                l += 1
            r += 1
        if ans == 10**9 + 1:
            return 0
        return ans

File: 209.cpp
#include <vector>
using namespace std;

class Solution {
public:
  int minSubArrayLen(int target, vector<int> &nums) {
    int l = 0, r = 0, curr = 0;
    int ans = __INT_MAX__;
    while (r < nums.size()) {
      curr += nums[r];
      while (curr >= target) {
        curr -= nums[l];
        ans = min(ans, r - l + 1);
        ++l;
      }
      ++r;
    }
    if (ans == __INT_MAX__) {
      return 0;
    }
    return ans;
  }
};